R/compute_CCpair_score.R

```diff
@@ -1,42 +1,102 @@
 #' compute_CCpair_score
 #'
-#' \code{compute_CCpair_score} computes the score for each cell - cell pair for all the patients.
+#' `compute_CCpair_score` computes the score for each cell-cell pair for all the patients.
 #'
 #' @export
 #'
 #' @param celltype1 string character with first cell type involved in the interaction
 #' @param celltype2 string character with second cell type involved in the interaction
-#' @param intercell.network matrix with data on cell types interaction network
-#' @param lrpairs.binary binary vector displaying LR pairs with non-zero frequency
-#' @param lr.frequency numeric vector with LR pairs frequency across the whole TCGA database
-#' @param compute.log boolean variable in order to take the log of the weighted score
-#' @param cancertype string character
+#' @param intercell_network matrix with data on cell types interaction network
+#' @param lrpairs_binary binary vector displaying LR pairs with non-zero frequency
+#' @param lr_frequency numeric vector with LR pairs frequency across the whole TCGA database
+#' @param compute_log boolean variable in order to take the log of the weighted score
 #'
 #' @return numeric vector with weighted scores
 #'
 #' @examples
-#' # TODOTODO
-compute_CCpair_score <- function(celltype1, celltype2, intercell.network, lrpairs.binary, lr.frequency, compute.log=TRUE, cancertype) {
+#' # Example: Mariathasan cohort (Mariathasan et al., Nature, 2018)
+#' if (!requireNamespace("BiocManager", quietly = TRUE))
+#'  install.packages("BiocManager")
+#'
+#' BiocManager::install(c("biomaRt",
+#'  "circlize",
+#'  "ComplexHeatmap",
+#'  "corrplot",
+#'  "DESeq2",
+#'  "dplyr",
+#'  "DT",
+#'  "edgeR",
+#'  "ggplot2",
+#'  "limma",
+#'  "lsmeans",
+#'  "reshape2",
+#'  "spatstat",
+#'  "survival",
+#'  "plyr"))
+#'
+#' install.packages("Downloads/IMvigor210CoreBiologies_1.0.0.tar.gz", repos = NULL)
+#' library(IMvigor210CoreBiologies)
+#'
+#' data(cds)
+#' mariathasan_data <- preprocess_mariathasan(cds)
+#' gene_tpm <- mariathasan_data$tpm
+#' rm(cds)
+#'
+#' # Computation of ligand-receptor pair weights
+#' lrpair_weights <- compute_LR_pairs(
+#'   RNA_tpm = gene_tpm,
+#'   remove_genes_ICB_proxies = FALSE,
+#'   cancer_type = "pancan")
+#'
+#' # remove ligand receptor pairs that are always NA
+#' na_lrpairs <- apply(lrpair_weights, 2, function(x) {
+#'  all(is.na(x))
+#'  })
+#' lrpair_weights <- lrpair_weights[, na_lrpairs == FALSE]
+#'
+#' # binarize the data: set a threshold to 10 TPM, only pairs where both ligand and receptor have TPM > 10 are kept
```

Comment:
split up this line - we should get to <100 characters, otherwise it is rendered potentially wrong in the manpages